File: src/main/java/org/ucl/newton/framework/Experiment.java
/*
 * Newton (c) 2018
 *
 * This work is licensed under the MIT License. To view a copy of this
 * license, visit
 *
 *      https://opensource.org/licenses/MIT
 */

package org.ucl.newton.framework;

/**
 * Instances of this class represent a research experiment, a data science
 * exercise conducted against a given set of data.
 *
 * @author Blair Butterworth
 */
public class Experiment
{
    private String id;
    private String name;

    public Experiment(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: application/master/src/main/webapp/resources/scripts/views/project/experiment.js
/**
 * Newton (c) 2018
 *
 * This work is licensed under the MIT License. To view a copy of this
 * license, visit
 *
 *      https://opensource.org/licenses/MIT
 *
 * Author: Blair Butterworth
 */

$(document).ready(function() {
    experimentExecuting = $('#experimentExecuting').val();
    if (experimentExecuting === 'true') {
        setTimeout(checkComplete, 2000);
    }
});

function checkComplete() {
    experimentId = $('#experimentId').val();
    $.ajax({
        type: 'GET',
        url: "/api/experiment/iscomplete?experimentId=" + experimentId,
        success: function(data) {
            if (data === true) {
                location.reload();
            }
            else {
                setTimeout(checkComplete, 2000);
            }
        },
    });
}